files/SZWESAMP/ZWEIMVS

```diff
@@ -0,0 +1,41 @@
+//ZWEIMVS JOB                                     
+//*                                                                    
+//* This program and the accompanying materials are made available     
+//* under the terms of the Eclipse Public License v2.0 which          
+//* accompanies this distribution, and is available at                 
+//* https://www.eclipse.org/legal/epl-v20.html                         
+//*                                                                    
+//* SPDX-License-Identifier: EPL-2.0                                   
+//*                                                                    
+//* Copyright Contributors to the Zowe Project. 2020, 2020             
+//*                                                                    
+//********************************************************************* 
+//*
+//* This job is used to create datasets used by a Zowe "instance"
+//* Instances represent a configuration of Zowe, different from the
+//* "runtime" datasets that are created upon install of Zowe / SMPE.
+//*
+//* If your choosen value of 'zowe.setup.dataset.authLoadlib' is not
+//* Equal to 'zowe.setup.prefix' + 'SZWELOAD',
+//* Then you must also run "ZWEIMVS2".
+//*
+//********************************************************************* 
+//MKPDSE EXEC PGM=IKJEFT01                       
+//SYSTSPRT DD SYSOUT=A                            
+//SYSTSIN DD *                                    
+ALLOC NEW DA('{zowe.setup.dataset.parmlib}') +       
+dsntype(library) dsorg(po) recfm(f b) lrecl(80) + 
+unit(sysallda) space(15,15) tracks                
+  
+ALLOC NEW DA('{zowe.setup.dataset.authPluginLib}') +      
+dsntype(library) dsorg(po) recfm(u) lrecl(0) +    
+blksize(32760) unit(sysallda) space(30,15) tracks 
+//*                                               
+//MEMBCPY EXEC PGM=IEBCOPY                         
+//SYSPRINT DD SYSOUT=A                            
+//SYSUT1 DD DSN={zowe.setup.dataset.prefix}.SZWESAMP,DISP=SHR  
+//SYSUT2 DD DSN={zowe.setup.dataset.parmlib},DISP=OLD
+//SYSIN DD *                                      
+  COPY OUTDD=SYSUT2,INDD=SYSUT1                                  
+  SELECT MEMBER=((ZWESIP00,,R))                          
```

Comment:
* `SELECT MEMBER=((ZWESIP00,{},R))` - does not allow customization
* `SELECT MEMBER=((ZWESIP00,{zowe.setup.dataset.parmlibMembers.zis},R))` - allows customization
  * Works fine if value is defined in yaml file
  * Does not work otherwise: `SELECT MEMBER=((ZWESIP00,CFG.ZOWE.SETUP.DATASET.PARMLIBMEMBERS.ZIS,R))`

Reply:
I have not found a perfect solution, but the way it works now seems good enough:
1) if you are working with JCL directly, you should be reviewing these, and adjusting them if you want.
2) if you are working with zwe instead, zwe edits this file to do a substitution.

I'm not sure how to, but if the rexx code had a parameter that would leave entries blank if not found then we would avoid having these `CFG.ZOWE.SETUP....` things in places we wont want them?

Reply:
1. Yes, no problem. We might add comment "If you change the suffix of parmlib member, update STC too." It is documented, but would be good to have it here.
2. This is out fault. We are providing way to customize `zowe.setup.dataset.parmlibMembers.zis`, so this customization must be complete, allowed here and in the STC for `ZWESIS01`.

Any solution between is confusing.

Reply:
I'm going to move ZWESIP00 over to defaults.yaml so that people are less likely to edit it. Not that they cannot, but I've never seen anyone do it and so it should be discouraged because it seems pointless to change?

Reply:
It is not so pointless to change, you can have ZWESIPAB for LPAR ABBA and ZWESIPKI for LPAR KISS. In case of sysplex and shared parmlib dataset, you have the only option to differ by suffix. Once the zowe will be used more, these changes will be more likely to happen. Anyway, move it to default for now.

Reply:
This is partial solution and it is will require manual updates from user.